### PR TITLE
Refactor OTLP mock collector test

### PR DESCRIPTION
**Pull request by alanwest**
Follow up from https://github.com/open-telemetry/opentelemetry-dotnet/pull/3788#issuecomment-1289805871

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                }))
+           .StartAsync();
+
+        var httpClient = host.GetTestClient();
+
+        var codes = new[] { Grpc.Core.StatusCode.Unimplemented, Grpc.Core.StatusCode.OK };
+        await httpClient.GetAsync($"/MockCollector/SetResponseCodes/{string.Join(",", codes.Select(x => (int)x))}");
+
+        var exportResults = new List<ExportResult>();
+        var otlpExporter = new OtlpTraceExporter(new OtlpExporterOptions() { HttpClientFactory = () => httpClient });
```
Comment (alanwest) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs:74]: This won't work today. But if #3393 is addressed first, then this test should go green.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                               var codes = responseCodesCsv.Split(",").Select(x => int.Parse(x)).ToArray();
+                               collectorState.SetStatusCodes(codes);
+                           });
+
+                       endpoints.MapGrpcService<MockTraceService>();
+                   });
+                }))
+           .StartAsync();
+
+        var httpClient = host.GetTestClient();
```
Comment (cijothomas) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs:68]: can we use a normal httpclient, not the Test "httpclient"?

Comment (alanwest) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs:68]: Yep, we could. That's what we do in the Grpc.Net.Client tests https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcServer.cs#L71.

Do we have a preference? I kinda like the TestServer/TestClient as I think it makes the test code more concise. That is, I think finding an open port, performing the ConfigureKestrel thing, etc. Not a huge deal, but it's extra noise.

Comment (alanwest) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/MockCollectorIntegrationTests.cs]: Ok, you twisted my arm 😄. Still planning on addressing #3393, but this PR is good to go now.

I hard-coded ports for now making sure to use something that doesn't conflict with our other test with a hard-coded port

https://github.com/open-telemetry/opentelemetry-dotnet/blob/be78372696cfdb4358854fdb32b859b778150661/test/OpenTelemetry.Instrumentation.AspNetCore.Tests/InProcServerTests.cs#L55

**Review by cijothomas — APPROVED** [review]

**Pull request merged**